app/routes.py

```diff
@@ -1,2 +1,218 @@
-from flask import Blueprint
+from app import db
+import requests
+from flask import request, Blueprint, make_response
+from flask import jsonify
+from app.models.customer import Customer
+from app.models.video import Video
+# from app.models.rental import Rental
+from dotenv import load_dotenv
+from datetime import datetime
+import os
+
+
+customers_bp = Blueprint("customers", __name__, url_prefix="/customers")
+videos_bp = Blueprint("videos", __name__, url_prefix="/videos")
+# rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+
+
+@customers_bp.route("", methods=["POST"], strict_slashes=False)
+def create_customer():
+    request_body = request.get_json()
+    
+    if ("name" not in request_body or "postal_code" not in request_body \
+        or "phone" not in request_body):
+        return jsonify({"erro":"Invalid data"}),400
+    
+    else:
+        customer = Customer(name = request_body["name"],
+                        postal_code = request_body["postal_code"],
+                        phone = request_body["phone"],
+                        registered_at = datetime.now())
+
+    db.session.add(new_customer)
+    db.session.commit()
+    return jsonify({
+        "id": customer.id
+    }), 201
+    
+
+@customers_bp.route("", methods=["GET"], strict_slashes=False)
+def get_customers():
+    customers = Customer.query.all()
+    
+    customer_list = []
+    for customer in customers:
+        customers_response.append(customer_list.to_json())
+    return jsonify(customer_list)
+
+
+@customers_bp.route("/<customer_id>", methods=["GET", "PUT", "DELETE"], strict_slashes=False)
+def retrieve_customer(customer_id):
+
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return "Customer not found.", 404
+
+    if request.method == "GET":
+        return customer.to_json_customer()
+    
+    if request.method == "PUT":
+
+        customer_data = request.get_json()
+
+        if "name" not in customer_data or "postal_code" not in customer_data or "phone" not in customer_data:
+            return jsonify({
+                "details": "Invalid data.  No name, postal code, and/or phone."
+            }), 400
+
+        customer.name = customer_data["name"]
+        customer.postal_code = customer_data["postal_code"]
+        customer.phone_num = customer_data["phone"]
+
+        db.session.commit()
+
+        return customer.to_json_customer()
+    
+    if request.method == "DELETE":
+
+        db.session.delete(customer)
+        db.session.commit()
+
+        return jsonify({
+            "id": customer.id})
```

Comment:
In the future, you'll encounter more detailed responses with a key providing a message like so:

```suggestion
        return jsonify({
            "id": customer.id,
            "message": f"Customer {customer.id} has been deleted."
            })
```